derive/src/max_encoded_len.rs

```diff
@@ -62,21 +59,32 @@ fn max_encoded_len_trait(input: &DeriveInput) -> syn::Result<TraitBound> {
 }
 
 // Add a bound `T: MaxEncodedLen` to every type parameter T.
-fn add_trait_bounds(mut generics: Generics, mel_trait: TraitBound) -> Generics {
+fn add_trait_bounds(
```

Comment:
Instead of this we should be able to use the `trait_bounds::add` function?

Reply:
We could, but I'm thinking whether it makes sense as it has quite some overhead, like calling `get_types_to_add_trait_bound`, whereas this one only needs a loop over `generics.params`.

Reply:
The `trait_bounds::add` function is just this method here on steroids. This one should have been using the `trait_bounds::add` from the beginning, but probably because of copying this code from substrate it wasn't done.